src/credentials/loginManager.ts

```diff
@@ -54,23 +55,30 @@ export class LoginManager {
                 accountId: accountId,
                 defaultRegion: provider.getDefaultRegion(),
             })
-        } catch (err) {
-            getLogger().error(
-                `Error trying to connect to AWS with Credentials Provider ${asString(
-                    args.providerId
-                )}. Toolkit will now disconnect from AWS. %O`,
-                err as Error
-            )
-            this.store.invalidateCredentials(args.providerId)
 
-            await this.logout()
+            return true
+        } catch (err) {
+            // TODO: implement custom exceptions instead of checking error message
```

Comment:
Probably not necessary, I don't think we'd handle these differently.

Reply:
It's more for when you want to change the text of an error message, but code exists that depends on the specific contents of the message. You see this a lot for tests. This could be solved by having a hidden 'type' field that would never change.